synopsys-task/src/synopsys-task/azure-service-client.ts

```diff
@@ -19,14 +19,17 @@ export class SynopsysAzureService {
       process.env["BUILD_REASON"] !== "PullRequest"
     ) {
       const StringFormat = (url: string, ...args: string[]) =>
-        url.replace(/{(\d+)}/g, (match, index) => args[index] || "");
+        url.replace(
+          /{(\d+)}/g,
+          (match, index) => encodeURIComponent(args[index]) || ""
+        );
 
       const endpoint = StringFormat(
         azureData.api.url.concat(this.azureGetMergeRequestsAPI),
         azureData.organization.name,
         azureData.project.name,
         azureData.repository.name,
-        "refs/heads/".concat(azureData.repository.branch.name),
+        azureData.repository.branch.name,
```

Comment:
azureData.repository.branch.name will already have refs/heads/ appended?

Reply:
yes, Both in PR and Non PR Context

Reply:
can we add some test cases to handle negative scenarios to make sure if something else come it has proper error handling.

Add debug logs for env variables or data that is not set directly by user but derived internally.

Reply:
updated test cases accordingly